src/bpfilter/xlate/cli.c

```diff
@@ -56,6 +60,206 @@ int _bf_cli_ruleset_flush(const struct bf_request *request,
     return bf_response_new_success(response, NULL, 0);
 }
 
+static void bf_nop_free(struct bf_chain **chain)
+{
+    UNUSED(chain);
+}
+
+static size_t _bf_cli_num_rules(bf_list *chain_list)
+{
+    bf_assert(chain_list);
+
+    size_t count = 0;
+    bf_list_foreach (chain_list, chain_node) {
+        struct bf_chain *chain = bf_list_node_get_data(chain_node);
+
+        count += bf_list_size(&chain->rules);
+    }
+
+    return count;
+}
+
+static int _bf_get_ctr_vals(bf_list *chain_list, struct bf_counter *counters)
+{
+    int counter_index = 0;
+    int r;
+
+    bf_assert(chain_list);
+    bf_assert(counters);
```

Comment:
I usually try to perform NULL checks together: `bf_assert(chain_list && counters)`.